.teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt

```diff
@@ -20,24 +20,35 @@ class IndividualPerformanceScenarioWorkers(model: CIBuildModel, os: Os = Os.linu
     applyPerformanceTestSettings(os = os, timeout = 420)
     artifactRules = individualPerformanceTestArtifactRules
 
+    if (os == Os.windows) {
+        // to avoid pathname too long error
+        vcs {
+            checkoutDir = "C:\\gradle-perf"
+        }
+    }
+
     params {
         param("baselines", "defaults")
         param("templates", "")
-        param("channel", "commits")
+        param("channel", if (os == Os.linux) "commits" else "commits-windows")
         param("checks", "all")
         param("runs", "defaults")
         param("warmups", "defaults")
         param("scenario", "")
 
-        param("env.ANDROID_HOME", "/opt/android/sdk")
-        param("env.PATH", "%env.PATH%:/opt/swift/4.2.3/usr/bin")
+        param("env.ANDROID_HOME", os.androidHome)
+        if (os == Os.linux) {
+            param("env.PATH", "%env.PATH%:/opt/swift/4.2.3/usr/bin")
+        } else {
+            param("env.PATH", "%env.PATH%;C:/Program Files/7-zip")
+        }
     }
 
     steps {
         script {
             name = "KILL_GRADLE_PROCESSES"
             executionMode = BuildStep.ExecutionMode.ALWAYS
-            scriptContent = killAllGradleProcesses
+            scriptContent = if (os == Os.windows) killAllGradleProcessesWindows else killAllGradleProcessesLinux
```

Comment:
Can this be a property of the `common.Os` type instead? So here we'd say:

```suggestion
            scriptContent = os.killAllGradleProcesses
```

Reply:
Done.